lib/inMemFileDB.js

```diff
@@ -66,26 +69,18 @@ class InMemoryFileDB {
 
         // path is always relative to appName.js-controller
         this.dataDir = (this.settings.connection.dataDir || getDefaultDataDir());
-        if (this.dataDir) {
-            if (this.dataDir[0] === '.' && this.dataDir[1] === '.') {
-                this.dataDir = __dirname + '/../' + this.dataDir;
-            } else if (this.dataDir[0] === '.' && this.dataDir[1] === '/') {
-                this.dataDir = __dirname + '/../' + this.dataDir.substring(2);
-            }
-        }
-        this.dataDir = path.normalize(this.dataDir);
+        this.dataDir = path.normalize(path.join(__dirname, '..', this.dataDir) + path.sep);
         this.dataDir = this.dataDir.replace(/\\/g, '/');
```

Comment:
Addinig path.sep and replacing \\ with / should also not be necessary if we use the path methods everywhere.

Reply:
The topic is that "this.dataDir" is coming from config and so can be any kind of path ... I would leave it as security net

Reply:
PS: original code https://github.com/ioBroker/ioBroker.js-controller/blob/master/lib/objects/objectsInMemServer.js#L44

So I changed it and removed the comment